CMakeLists.txt

```diff
@@ -392,9 +392,9 @@ else()
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DGTEST_HAS_TR1_TUPLE=0")
   endif()
   if(CMAKE_C_COMPILER_ID STREQUAL "GNU" AND NOT (CMAKE_C_COMPILER_VERSION VERSION_LESS 4.8))
-    set(DEBUG_FLAGS "-g3 -Og")
+    set(DEBUG_FLAGS "-g3 -Og -Wall -fprofile-arcs -ftest-coverage --coverage")
   else()
-    set(DEBUG_FLAGS "-g3 -O0")
+    set(DEBUG_FLAGS "-g3 -O0 -Wall -fprofile-arcs -ftest-coverage --coverage")
```

Comment:
There is logic elsewhere that appends -Wall to *FLAGS, see `WARNINGS` var. Could we avoid duplication? If somehow -Wall doesn't end up where you want it, let's investigate why.

Agreed on either a separating profiling. How about add a cmake option (with `option()`) for this and set it to false by default and set it to true in a new 'profile' target in top-level makefile?